core/domain/user_services.py

```diff
@@ -1908,9 +1901,12 @@ def get_human_readable_user_ids(user_ids):
     usernames = []
     for ind, user_settings in enumerate(users_settings):
         if user_settings is None:
-            logging.error('User id %s not known in list of user_ids %s' % (
-                user_ids[ind], user_ids))
-            raise Exception('User not found.')
+            if strict:
+                logging.error('User id %s not known in list of user_ids %s' % (
+                    user_ids[ind], user_ids))
+                raise Exception('User not found.')
+            else:
+                usernames.append(USER_IDENTIFICATION_FOR_USER_BEING_DELETED)
```

Comment:
Same note as above, about why we need this (though I think what you've done here makes more sense in this context because something that's "human readable" is unlikely to be parsed further).

Again please update the Returns docstring to be clearer, though.

Reply:
Done.